### PR TITLE
jquery added, ready to push to origin

### PR DESCRIPTION
**Pull request by cxoltero**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -1,115 +1,32 @@
-(function() {
+$(document).ready(function(){
     "use strict";
 
-    function loadPics() {
```
Comment (paul-barry-kenzan) [inline js/main.js:4]: Sorry, I may have been misleading. The parent IIFE should remain, the inner one is the one I'd like to replace with `$(document).ready()`.

``` javascript
(function($){ // <- The $ here is our custom alias to the jQuery global included below.
  $(document).ready(function(){
    // app code here
  });
})(jQuery); // <- This is how we gain access of globals inside our IIFE, jQuery in this case

```

File: js/main.js [hunk 2]
```diff
@@ -1,115 +1,32 @@
...
-    function pupulate(pics){
-        for(var j=0; j<pics.length; j++){
-            var list = document.getElementById('list');
-            for(j = 0; j<pics.length; j++){
-                list.innerHTML = list.innerHTML + "<li class= 'active' id='img"+[j]+"'><img src='" + pics[j] + "' /></li>";
+        function createDomElm(data){
+            var farm, server, id, title, secret, picUrl, img;
+
+            for(var i=0; i< data.photos.photo.length; i++){
+                farm = data.photos.photo[i].farm;
```
Comment (paul-barry-kenzan) [inline js/main.js]: To clean this up a bit, let's do something like this:

``` javascript
var currentPhoto;
for(var i=0; i< data.photos.photo.length; i++){
  currentPhoto = data.photos.photo[i]; // Create a simple reference to increase readability

  farm = currentPhoto.farm;
  server = currentPhoto.server;
  // ... remaining data values
}
```

File: js/main.js [hunk 3]
```diff
@@ -1,115 +1,34 @@
...
-            pics.push(picUrl);
-        }
-    }
-    function pupulate(pics){
-        for(var j=0; j<pics.length; j++){
-            var list = document.getElementById('list');
-            for(j = 0; j<pics.length; j++){
-                list.innerHTML = list.innerHTML + "<li class= 'active' id='img"+[j]+"'><img src='" + pics[j] + "' /></li>";
+    $(document).ready(function(){
+        (function loadPics(){
```
Comment (paul-barry-kenzan) [inline js/main.js]: Shouldn't need the IIFE here. 

``` javascript
(function($){

  $(document).ready(function(){
    loadPics();
  });

  // OR
  // could try this
  // $(document).ready(loadPics); 

  function loadPics(){
    // function code here
  }

})(jQuery);
```

File: js/main.js [hunk 4]
```diff
@@ -1,115 +1,34 @@
...
+
+                for(var i=0; i< data.photos.photo.length; i++){
+                    currentPhoto = data.photos.photo[i];
+
+                    farm = currentPhoto.farm;
+                    server = currentPhoto.server;
+                    id = currentPhoto.id;
+                    title = currentPhoto.title;
+                    secret = currentPhoto.secret;
+                    picUrl = "https://farm"+farm+".staticflickr.com/"+server+"/"+id+"_"+secret+"_z.jpg";
```
Comment (paul-barry-kenzan) [inline js/main.js]: Great! This looks much better. I like to do this anytime I have to reference by an index, and especially when there are so many child objects: `data.photos.photo[i].farm;`

Comment (cxoltero) [inline js/main.js]: Yeah it crossed my mind when i was doing it, I guess just wanted to try brute force first.  should i merge PR at this point?

File: js/main.js (hunk 3 above)
Comment (paul-barry-kenzan) [inline js/main.js]: @cxoltero 
Just this last update before merge.

**Pull request merged**